src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -636,11 +637,14 @@ public void Copy(Operation<T> operation, T objectToApplyTo)
             }
         }
 
-        private bool IsNonStringArray(JsonPatchProperty patchProperty)
+        private bool IsNonStringArray(Type type)
         {
-            return !(patchProperty.Property.PropertyType == typeof(string))
-                    && typeof(IList).GetTypeInfo().IsAssignableFrom(
-                        patchProperty.Property.PropertyType.GetTypeInfo());
+            if (PropertyHelpers.GetIListType(type) != null)
```

Comment:
If this is only used here please just inline the code. We don't want to litter our code with random helpers when a private method does what you want already.